### PR TITLE
feat: NTRN-262. Add interhchain queries Init/Export Genesis

### PR DESCRIPTION
**Pull request by albertandrejev**
Implement `ExportGenesis` and `InitGenesis` methods for Interchain Queries module in Neutron.

Registered Interchain Queries and other necessary info must be exported and initialised in the corresponding methods. Otherwise we won’t be able to upgrade our chain properly.

### CODE REVIEW

File: x/interchainqueries/genesis.go [hunk 1]
```diff
@@ -10,6 +10,14 @@ import (
 // InitGenesis initializes the capability module's state from a provided genesis
 // state.
 func InitGenesis(ctx sdk.Context, k keeper.Keeper, genState types.GenesisState) {
+	// Set all registere queries
```
Comment (mmikev) [inline x/interchainqueries/genesis.go]: registere**d**

File: x/interchainqueries/genesis.go [hunk 2]
```diff
@@ -10,6 +10,14 @@ import (
...
+	if genState.RegisteredQueries != nil {
```
Comment (mmikev) [inline x/interchainqueries/genesis.go]: Unnecessary check since you can safely iterate through nil slice

File: x/interchainqueries/genesis.go [hunk 3]
```diff
@@ -10,6 +10,14 @@ import (
...
+		for _, elem := range genState.RegisteredQueries {
+			k.SetLastRegisteredQueryKey(ctx, elem.Id)
+			k.SaveQuery(ctx, *elem)
```
Comment (mmikev) [inline x/interchainqueries/genesis.go]: Unhandled error

File: x/interchainqueries/keeper/keeper.go [hunk 4]
```diff
@@ -107,6 +107,25 @@ func (k Keeper) GetQueryByID(ctx sdk.Context, id uint64) (*types.RegisteredQuery
 	return &query, nil
 }
 
+// GetAllReigsteredQueries returns all registered queries
+func (k Keeper) GetAllReigsteredQueries(ctx sdk.Context) []*types.RegisteredQuery {
```
Comment (mmikev) [inline x/interchainqueries/keeper/keeper.go]: Re**gi**stered

**Review by pr0n00gler — CHANGES_REQUESTED** [review]

**Review by zavgorodnii — APPROVED** [review]

**Review by pr0n00gler — APPROVED** [review]

**Pull request merged**